zipline/algorithm.py

```diff
@@ -563,13 +563,21 @@ def _create_generator(self, sim_params):
         if sim_params is not None:
             self.sim_params = sim_params
 
+        benchmark_source = self._create_benchmark_source()
+
         if self.perf_tracker is None:
             # HACK: When running with the `run` method, we set perf_tracker to
             # None so that it will be overwritten here.
+            portfolio = zp.Portfolio(
+                data_portal=self.data_portal,
+                current_dt_callback=self.get_datetime,
+                benchmark_asset=benchmark_source.benchmark_asset,
+            )
             self.perf_tracker = PerformanceTracker(
```

Comment:
As far as I can tell, if we construct our own `PerformanceTracker`, the check for `if self.perf_tracker is None:` will prevent this code path from ever being hit then, meaning I think this is actually ok?